### PR TITLE
Fix boot_out.txt contents check

### LINKED ISSUE #5514: Code gets wiped when power-cycling between 2.8 and 3.0 seconds
### CircuitPython version

```python
Adafruit CircuitPython 7.0.0 on 2021-09-20; Raspberry Pi Pico with rp2040
```

### Code/REPL

```python
import board
import time
import digitalio

# variables
forever = 1

# main program
def main():
    heartbeat_toggle = False
    print("Hello")
    boardled = digitalio.DigitalInOut(board.GP25)
    boardled.direction = digitalio.Direction.OUTPUT
    out0 = digitalio.DigitalInOut(board.GP0)
    out0.direction = digitalio.Direction.OUTPUT
    while forever:
        heartbeat_toggle = not (heartbeat_toggle)
        boardled.value = heartbeat_toggle  # toggle the LED
        out0.value = heartbeat_toggle  # toggle GPIO0
        time.sleep(0.1)

main()
```

### Behavior

Hello, seeing an issue with the official 7.0 release of CircuitPython on Pi Pico.

It seems to handle power-cycling quite badly.. the chances are high that when a user plugs/unplugs a battery or controls a power switch, that the user's code.py could be wiped from Flash. There's at least a 200msec window where this happens. 

Also, it takes a very long time (just over 3 seconds) for the code.py to start executing. 

I only investigated because it happened several times to me within a week, just connecting/disconnecting a Battery with LDO regulator, and noticing my code was gone. 

To test it, I stripped down my code.py to just an LED blink program, and then applied power under control of a script to switch the power on/off on the VSYS pin.  It seems to only be an issue in 7.0. I tried the older 6.2 release and that behaves as expected, with a 1-second delay before code.py execution. I could not get that version 6.2 to fail with power cycling at 100msec increments between 100msec and 1 seconds. However with version 7.0, the startup time is just slightly over 3 seconds and I tried power cycling version 7.0 and between 2.8 seconds and 3.0 seconds the code gets wiped. I recorded a private unlisted YouTube video here showing the issue:

https://youtu.be/eG4jziJHroE

### Description

CircuitPython 7.0 startup time to execute code.py is unusually long (just over 3 seconds), and if a power-cycle occurs just before then, at 2.8-3.0 seconds, then the code gets wiped and the default code.py runs. The user code is permanently gone and then needs transferring again via USB.

The period 2.8-3.0 seconds might sound very specific, but I randomly got the problem to occur by just plugging a battery + LDO voltage regulator on/off the Pico board, occurring for me several times within a week just by accident. 

I have tested with two Pi Pico boards, and they both had the same issue. 

Also, the 3-second delay before code.py execution will cause issues because kids are not expecting to wait so long between pushing a power button on, and using their Pi Pico projects. That then causes a high probability that they will power-cycle within that time, a bit like pressing a button on a TV remote repeatedly if it doesn't respond in the expected time.

### Additional information

Using release 6.2 resolves the issue (startup time is reduced from 3 seconds to 1 second, and the code wiping issue has not been observed).

### PR DESCRIPTION
**Pull request by dhalbert**
- Fixes #5514.

`boot_out.txt` was getting rewritten every time, because the contents check did not include checking for the new `Board ID:` info.

I checked to see whether a longer concatenated string or two string comparisons was smaller in terms of firmware space. The latter was better.

I also reduced the time spent waiting for stable power from 1.5 seconds to 1 second.

Note that the existence of a `boot.py` will cause `boot_out.txt` to be written every time, since we don't know what will get written. It could get buffered instead, and then compared, to save flash writes, but that's more complicated.

Anybody can review.

### CODE REVIEW

File: extmod/moduselect.c [hunk 1]
```diff
@@ -150,7 +150,7 @@ STATIC mp_obj_t select_select(size_t n_args, const mp_obj_t *args) {
             mp_map_deinit(&poll_map);
             return mp_obj_new_tuple(3, list_array);
         }
-        MICROPY_EVENT_POLL_HOOK
+        RUN_BACKGROUND_TASKS;
```
Comment (tannewt) [inline extmod/moduselect.c:153]: These changes look to be async related.

**Review by tannewt — CHANGES_REQUESTED** [review]
Did you intend to include async changes here?

File: extmod/moduselect.c (hunk 1 above)
Comment (dhalbert) [reply extmod/moduselect.c:153]: No, this was a git mess-up. I added only certain files, but must have missed some.

**Comment by dhalbert** [comment]
Force-pushed correct changes.

**Comment by tannewt** [comment]
This PR will likely be closed in favor of #5536

**Pull request closed without merging**